### PR TITLE
Added how-to guide: convert pure pytorch code to ignite

### LINKED ISSUE #14: Convert pure pytorch code to ignite
Following on the discussion with @vfdev-5 , this new how-to guide should be abstract and:
> The purpose is to show explicitly how to convert pure pytorch code to ignite and explain what we gain by that

Topics to cover:
- Training loop
- Validation loop
- Metrics
- Checkpoints
- LR Scheduler

### PR DESCRIPTION
**Pull request by Priyansi**
Fixes #14 

Abstract code comparing PyTorch and Ignite for:
- [x] Training loop
- [x] Validation loop
- [x] Metrics
- [x] Events and Handlers
- [x] Progress Bar
- [x] Checkpoint

To do:
- [x] GIF showing moving code once the notebook is finalized

### CODE REVIEW

File: how-to-guides/convert-pytorch-to-ignite.ipynb [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+      "execution_count": 8,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "WnGK925N5AR7"
+      },
+      "source": [
+        "## Organising code into Events and Handlers\n",
```
Comment (vfdev-5) [inline how-to-guides/convert-pytorch-to-ignite.ipynb:228]: Organising vs Organizing, I imagine it is British vs American. Let's use "z" as otherwise looks like a typo. What do you think ?

File: how-to-guides/convert-pytorch-to-ignite.ipynb [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "sRgDrTgi5AU_"
+      },
+      "source": [
+        "### Progress Bar\n",
+        "\n",
+        "Instead of `tqdm`, we can use the built-in [`ProgressBar()`](https://pytorch.org/ignite/generated/ignite.contrib.handlers.tqdm_logger.html#module-ignite.contrib.handlers.tqdm_logger)."
```
Comment (vfdev-5) [inline how-to-guides/convert-pytorch-to-ignite.ipynb:302]: > Instead of `tqdm`, we can use the built-in ProgressBar()

ProgressBar is a wrapper upon tqdm, so this sentence seems a bit strange...

File: how-to-guides/convert-pytorch-to-ignite.ipynb [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+      "metadata": {
+        "id": "vkqMcVnA5AZ3"
+      },
+      "source": [
+        "### Checkpointing\n",
+        "\n",
+        "Instead of saving all models after `checkpoint_every` epoch:\n",
+        "```python\n",
+        "if epoch % checkpoint_every == 0:\n",
+        "    checkpoint(model, optimizer, \"checkpoint_directory\")\n",
```
Comment (vfdev-5) [inline how-to-guides/convert-pytorch-to-ignite.ipynb:329]: Here you save model and optimizer but in ignite code only model

**Review by vfdev-5 — COMMENTED** [review]
@Priyansi nice how to guide. Few improvements to bring and it is good to go.
I remember also we wanted to make and insert here a gif.
Maybe, it would worth to add few links (something like complete code example) in the end.

**Comment by vfdev-5** [comment]
@Priyansi thanks for updates.

Let's add few links in the end of the text on a complete code example

File: how-to-guides/convert-pytorch-to-ignite.ipynb [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+      "source": [
+        "from ignite.handlers import Checkpoint, DiskSaver\n",
+        "\n",
+        "checkpoint_every = 5\n",
+        "checkpoint_dir = ...\n",
+        "\n",
+        "\n",
+        "checkpointer = Checkpoint(\n",
+        "    to_save={'model': model, 'optimizer': optimizer, 'trainer': trainer}, save_handler=DiskSaver(checkpoint_dir, create_dir=True), n_saved=2\n",
+        ")\n",
```
Comment (vfdev-5) [inline how-to-guides/convert-pytorch-to-ignite.ipynb:358]:
```suggestion
        "    to_save={'model': model, 'optimizer': optimizer, 'trainer': trainer},\n",
        "    save_handler=DiskSaver(checkpoint_dir, create_dir=True), n_saved=2\n",
        ")\n",
```

**Review by vfdev-5 — APPROVED** [review]
A nit fix, otherwise looks good to merge

**Comment by vfdev-5** [comment]
I hope GIF rendering will be neater than on GH

**Comment by Priyansi** [comment]
@vfdev-5 i made the suggested change.
